src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -167,4 +187,39 @@ private void updateNonFormProperties(GivrUser user, boolean forRegistration, Emb
     
     this.entity.setProperty("lastEditTimeStampMillis", millisecondSinceEpoch);
   }
+
+  private void updateOpenHoursProperty(HttpServletRequest request) {
+    ArrayList<EmbeddedEntity> hoursOpen = new ArrayList<EmbeddedEntity>();
+    ArrayList<String> dayOptionFromTimes = new ArrayList<String>();
+    ArrayList<String> dayOptionToTimes = new ArrayList<String>();
+
+    for (DayOfWeek currDay : DayOfWeek.values()) {
+      EmbeddedEntity dayOption = new EmbeddedEntity();
+      dayOption.setProperty("day", currDay.toString());
+      String isOpen = getParameterOrThrow(request, currDay.toString() + "-isOpen");
+      if(isOpen.equals("open")) {
+        dayOption.setProperty("isOpen", "true");
+        dayOptionFromTimes = getParameterValuesOrThrow(request, currDay.toString() + "-from-times");
+        dayOptionToTimes = getParameterValuesOrThrow(request, currDay.toString() + "-to-times");
+        ArrayList<EmbeddedEntity> fromToPairs = createFromToPairs(dayOptionFromTimes, dayOptionToTimes);
+        dayOption.setProperty("fromToPairs", fromToPairs);
+      } else {
+        dayOption.setProperty("isOpen", false);
+      }
+      hoursOpen.add(dayOption);
+    }
+
+    this.entity.setProperty("orgHoursOpen", hoursOpen);
+  }
+
+  private ArrayList<EmbeddedEntity> createFromToPairs(ArrayList<String> dayOptionFromTimes, ArrayList<String> dayOptionToTimes) {
+    ArrayList<EmbeddedEntity> pairs = new ArrayList<EmbeddedEntity>();
+    for(int i = 0; i < dayOptionFromTimes.size(); i++) {
+      EmbeddedEntity fromToPair = new EmbeddedEntity();
+      fromToPair.setProperty("from", dayOptionFromTimes.get(i));
+      fromToPair.setProperty("to", dayOptionToTimes.get(i));
```

Comment:
The input type used only allows a time to be inputted. If from > to then it would be a user error but that is also why we give them the option to edit their organization data

Reply:
You're right that it's a user error, but in general it's important to have a good understanding of the format of the data in your datastore. (Which means validating it!) It's important to consider:
 - data you don't expect may come from malicious users as well! 
 - even if the use of the data is safe currently, future developers (including yourself) may make assumptions about the format of the data based on the common case and not consider that time might be "seven" or "<script>"

That all said, this seems like we can separate it from the rest of this change! What do you think about creating a github issue to track a fix for this, and addressing it in a follow up PR?

Reply:
Got it! I will create an issue and then address it